src/data-workspace/data-details-sidebar/comment.js

```diff
@@ -83,17 +93,44 @@ const CommentEditField = ({ comment, commentOptionSetId }) => {
 
 CommentEditField.propTypes = {
     comment: PropTypes.string,
+    commentId: PropTypes.string,
     commentOptionSetId: PropTypes.string,
+    unsavedComment: PropTypes.string,
+    onBlur: PropTypes.func,
 }
 
-function CommentEditForm({ item, open, setOpen, syncComment, closeEditor }) {
+function CommentEditForm({
+    item,
+    open,
+    setOpen,
+    syncComment,
+    closeEditor,
+    unsavedComment,
+    commentId,
+}) {
+    const contextSelectionId = useContextSelectionId()
+    const cellId = getCellId({ contextSelectionId, item })
+    const setUnsavedComment = useUnsavedDataStore(
+        (state) => state.setUnsavedComment
+    )
+
     const onSubmit = (values) => {
         // Don't send `undefined` (or 'undefined' will be stored as the comment)
-        const comment = values.comment || ''
+        const comment = values[`comment_${commentId}`] || ''
         syncComment({ comment })
+        setUnsavedComment(cellId, null)
```

Comment:
Should we have a `deleteUnsavedComment` instead? That way we are not "persisting" the ID at all. I'm pretty sure this would still linger in the store and not be garbage collected.

```
deleteUnsavedComment: (cellId) =>
    set((state) => {
        const unsavedComments = { ...state.unsavedComments }
        delete unsavedComments[cellId]
        return { unsavedComments }
    }),```